src/main/java/seedu/pill/util/Parser.java

```diff
@@ -63,8 +64,17 @@ public void parseCommand(String input) {
                 new HelpCommand(argument, flag).execute(this.items, this.storage);
                 break;
             case "list":
+                if (splitInput.length > 1) {
+                    throw new PillException(ExceptionMessages.TOO_MANY_ARGUMENTS);
+                }
```

Comment:
Good job adding more guard clauses, but could this be abstracted into another method?